plugins/ssh-plugin/src/ssh-plugin-backend.ts

```diff
@@ -82,24 +83,57 @@ const getSshServiceName = async function (): Promise<string> {
     }
 };
 
+const updateConfig = function (host: string): void {
+    const sshDir = os.homedir() + '/.ssh';
+    const configFile = sshDir + '/config';
+    if (!fs.existsSync(configFile)) {
+        if (!fs.existsSync(sshDir)) {
+            fs.mkdirSync(sshDir);
+        }
+        fs.appendFileSync(configFile, '');
+        fs.chmodSync(configFile, '644');
+    }
+    const keyConfig = `\nHost ${host}\nHostName ${host}\nIdentityFile ${sshDir}/${host.replace('.', '_')}\n`;
+    const configContent = fs.readFileSync(configFile).toString();
+    if (configContent.indexOf(keyConfig) >= 0) {
+        const newConfigContent = configContent.replace(keyConfig, '');
+        fs.writeFileSync(configFile, newConfigContent);
+    } else {
+        fs.appendFileSync(configFile, keyConfig);
+    }
+};
+
+const writeKey = function (name: string, key: string): void {
+    const keyFile = os.homedir() + '/.ssh/' + name.replace('.', '_');
+    fs.appendFileSync(keyFile, key);
+    fs.chmodSync(keyFile, '600');
+};
+
 const generateKeyPair = async function (sshkeyManager: SshKeyManager): Promise<void> {
-    const keyName = await theia.window.showInputBox({ placeHolder: 'Please provide a key pair name' });
+    const keyName = await theia.window.showInputBox({ placeHolder: 'Please provide a key pair name (Host name e.g. github.com)' });
```

Comment:
Are we prompting the user for a key pair name or for an host name?

Reply:
And do we really need to prompt the user for this? Can't we use a default name?

Reply:
> Are we prompting the user for a key pair name or for an host name?

Changed to `Host name`
> And do we really need to prompt the user for this? Can't we use a default name?

We need to distinguish keys by different git providers. We may have a key for github and bitbucket as well.

Reply:
I don't understand. The same key can be used with multiple git providers. 

When you generate a key pair, the public key is for sharing with everyone. There is no issue with sharing the same public key with multiple organisations.

Reply:
Agree, but the SSH plugin is initially designed to use multiple keys. basically the model is taken from che 6 https://www.eclipse.org/che/docs/che-6/version-control.html#using-existing-ssh-keys. To use single ssh key we need to completely rework the plugin and the way of restoring the ssh key on workspace start

Reply:
I don't want to block this PR but I still can't understand why, as a user, I should provide an hostname (i.e. github.com). I don't do that on my host when I setup my SSH key. Why I need that on a Che workspace?